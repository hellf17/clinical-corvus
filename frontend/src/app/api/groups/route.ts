import { auth } from '@clerk/nextjs/server';
import { NextRequest, NextResponse } from 'next/server';

// Simple in-memory rate limiting store (in production, use Redis or similar)
const rateLimitStore = new Map<string, { count: number; timestamp: number }>();

// Rate limiting configuration
const RATE_LIMIT_WINDOW = 60 * 1000; // 1 minute
const RATE_LIMIT_MAX_REQUESTS = 100; // 100 requests per window

// Rate limiting function
function isRateLimited(userId: string): boolean {
  const now = Date.now();
  const key = `${userId}`;
  const record = rateLimitStore.get(key);

  if (!record || now - record.timestamp > RATE_LIMIT_WINDOW) {
    // Reset rate limit counter
    rateLimitStore.set(key, { count: 1, timestamp: now });
    return false;
  }

  if (record.count >= RATE_LIMIT_MAX_REQUESTS) {
    // Rate limit exceeded
    return true;
  }

  // Increment counter
  rateLimitStore.set(key, { count: record.count + 1, timestamp: record.timestamp });
  return false;
}

export async function GET(request: NextRequest) {
  try {
    const { getToken } = await auth();
    const token = await getToken();
    
    if (!token) {
      return NextResponse.json(
        { error: 'Authentication required' },
        { status: 401 }
      );
    }

    // Rate limiting using token as user identifier
    if (isRateLimited(token)) {
      return NextResponse.json(
        { error: 'Rate limit exceeded' },
        { status: 429 }
      );
    }
    
    // Extract query parameters from the request URL
    const { searchParams } = new URL(request.url);
    const queryString = searchParams.toString();
    const url = queryString
      ? `${process.env.NEXT_PUBLIC_BACKEND_URL}/api/groups/?${queryString}`
      : `${process.env.NEXT_PUBLIC_BACKEND_URL}/api/groups/`;
    
    // Forward to backend
    const response = await fetch(url, {
      method: 'GET',
      headers: {
        'Content-Type': 'application/json',
        'Authorization': `Bearer ${token}`,
      },
    });

    if (!response.ok) {
      return NextResponse.json(
        { error: 'Failed to fetch groups' },
        { status: response.status }
      );
    }

    const data = await response.json();
    return NextResponse.json(data);
  } catch (error) {
    console.error('Error in GET /api/groups:', error);
    return NextResponse.json(
      { error: 'Internal server error' },
      { status: 500 }
    );
  }
}

export async function POST(request: NextRequest) {
  try {
    const { getToken } = await auth();
    const token = await getToken();
    
    if (!token) {
      return NextResponse.json(
        { error: 'Authentication required' },
        { status: 401 }
      );
    }

    // Rate limiting using token as user identifier
    if (isRateLimited(token)) {
      return NextResponse.json(
        { error: 'Rate limit exceeded' },
        { status: 429 }
      );
    }
    
    const body = await request.json();
    
    // Forward to backend
    const response = await fetch(`${process.env.NEXT_PUBLIC_BACKEND_URL}/api/groups/`, {
      method: 'POST',
      headers: {
        'Content-Type': 'application/json',
        'Authorization': `Bearer ${token}`,
      },
      body: JSON.stringify(body),
    });

    if (!response.ok) {
      return NextResponse.json(
        { error: 'Failed to create group' },
        { status: response.status }
      );
    }

    const data = await response.json();
    return NextResponse.json(data);
  } catch (error) {
    console.error('Error in POST /api/groups:', error);
    return NextResponse.json(
      { error: 'Internal server error' },
      { status: 500 }
    );
  }
}