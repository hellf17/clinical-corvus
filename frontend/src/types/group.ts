/**
 * Types for group management
 */

export type GroupRole = 'admin' | 'member';

export interface Group {
  id: number;
  name: string;
  description?: string;
  max_patients?: number;
  max_members?: number;
  created_at: string;
  updated_at: string;
}

export interface GroupMembership {
  id: number;
  group_id: number;
  user_id: number;
  role: GroupRole;
  joined_at: string;
  invited_by?: number;
}

export interface GroupPatient {
  id: number;
  group_id: number;
  patient_id: number;
  assigned_at: string;
  assigned_by?: number;
}

export interface GroupWithMembersAndPatients extends Group {
  members: GroupMembership[];
  patients: GroupPatient[];
}

export interface GroupCreate {
  name: string;
  description?: string;
  max_patients?: number;
  max_members?: number;
}

export interface GroupUpdate {
  name?: string;
  description?: string;
  max_patients?: number;
  max_members?: number;
}

export interface GroupMembershipCreate {
  user_id: number;
  role: GroupRole;
}

export interface GroupMembershipUpdate {
  role: GroupRole;
}

export interface GroupPatientCreate {
  patient_id: number;
}

export interface GroupWithCounts extends Group {
  member_count?: number;
  patient_count?: number;
}

export interface GroupListResponse {
  items: Group[];
  total: number;
}

export interface GroupWithCountsListResponse {
  items: GroupWithCounts[];
  total: number;
}

export interface GroupMembershipListResponse {
 items: GroupMembership[];
  total: number;
}

export interface GroupPatientListResponse {
  items: GroupPatient[];
  total: number;
}