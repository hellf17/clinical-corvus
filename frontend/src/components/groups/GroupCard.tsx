'use client';

import React from 'react';
import { GroupWithCounts } from '@/types/group';
import { Card, CardContent, CardHeader, CardTitle } from '@/components/ui/Card';
import { Button } from '@/components/ui/Button';
import { Users, Calendar, ChevronRight, UserCheck, Stethoscope } from 'lucide-react';
import Link from 'next/link';
import { format } from 'date-fns';
import { ptBR } from 'date-fns/locale';

interface GroupCardProps {
  group: GroupWithCounts;
  isSelected?: boolean;
  onSelect?: (groupId: string) => void;
}

export const GroupCard: React.FC<GroupCardProps> = ({ group, isSelected = false, onSelect }) => {
  const utilizationRate = {
    doctors: group.max_members ? Math.round(((group.member_count || 0) / group.max_members) * 100) : null,
    patients: group.max_patients ? Math.round(((group.patient_count || 0) / group.max_patients) * 100) : null
  };

  return (
    <Card
      className={`group transition-all duration-300 cursor-pointer h-full overflow-hidden ${
        isSelected
          ? 'shadow-2xl border-0 bg-gradient-to-br from-blue-50 via-indigo-50 to-purple-50 ring-2 ring-blue-500 ring-opacity-30'
          : 'hover:shadow-xl hover:scale-[1.02] shadow-lg border-0 bg-gradient-to-br from-white via-gray-50 to-blue-50/30 backdrop-blur-sm hover:ring-2 hover:ring-gray-300/50'
      }`}
      onClick={() => onSelect && onSelect(group.id.toString())}
    >
      {/* Header Section with Enhanced Gradient */}
      <CardHeader className={`pb-4 relative overflow-hidden ${
        isSelected
          ? 'bg-gradient-to-r from-blue-600 via-indigo-600 to-purple-600'
          : 'bg-gradient-to-r from-gray-800 via-gray-700 to-gray-800 group-hover:from-blue-700 group-hover:via-indigo-700 group-hover:to-purple-700'
      }`}>
        <div className="absolute inset-0 bg-black/10"></div>
        <CardTitle className="relative flex justify-between items-start">
          <div className="flex-1">
            <h3 className="text-xl font-bold mb-2 text-white drop-shadow-md">
              {group.name}
            </h3>
            {group.description && (
              <p className="text-sm line-clamp-2 text-white/90 leading-relaxed">
                {group.description}
              </p>
            )}
          </div>
          {isSelected && (
            <div className="flex-shrink-0 w-8 h-8 bg-white/20 backdrop-blur-sm rounded-full flex items-center justify-center ml-3 ring-2 ring-white/30">
              <div className="w-3 h-3 bg-white rounded-full animate-pulse"></div>
            </div>
          )}
        </CardTitle>
      </CardHeader>
      
      <CardContent className="pt-6 space-y-6 relative">
        {/* Enhanced Stats Section */}
        <div className="grid grid-cols-2 gap-4">
          {/* Doctors Card */}
          <div className={`relative overflow-hidden rounded-xl p-4 transition-all duration-300 ${
            isSelected
              ? 'bg-gradient-to-br from-blue-100 via-blue-50 to-indigo-100 border-2 border-blue-200/50 shadow-lg'
              : 'bg-gradient-to-br from-gray-50 via-white to-blue-50/30 border border-gray-200 hover:border-blue-300 hover:shadow-md group-hover:bg-gradient-to-br group-hover:from-blue-50 group-hover:via-white group-hover:to-indigo-50'
          }`}>
            <div className="absolute inset-0 bg-gradient-to-br from-white/60 via-transparent to-transparent"></div>
            <div className="relative flex items-center">
              <div className={`p-2 rounded-lg ${
                isSelected ? 'bg-blue-600 text-white' : 'bg-blue-100 text-blue-600 group-hover:bg-blue-600 group-hover:text-white'
              } transition-all duration-300`}>
                <Stethoscope className="h-5 w-5" />
              </div>
              <div className="ml-3 flex-1">
                <p className={`text-xs font-semibold uppercase tracking-wide ${
                  isSelected ? 'text-blue-600' : 'text-gray-500 group-hover:text-blue-600'
                }`}>
                  Médicos
                </p>
                <div className="flex items-baseline space-x-1">
                  <p className={`text-2xl font-black ${
                    isSelected ? 'text-blue-800' : 'text-gray-900 group-hover:text-blue-800'
                  }`}>
                    {group.member_count || 0}
                  </p>
                  {group.max_members && (
                    <p className="text-sm text-gray-500">
                      /{group.max_members}
                    </p>
                  )}
                </div>
                {utilizationRate.doctors && (
                  <div className="mt-2">
                    <div className="flex items-center justify-between mb-1">
                      <span className="text-xs text-gray-500">Ocupação</span>
                      <span className="text-xs font-semibold text-blue-600">{utilizationRate.doctors}%</span>
                    </div>
                    <div className="w-full bg-gray-200 rounded-full h-1.5">
                      <div
                        className="bg-gradient-to-r from-blue-500 to-indigo-500 h-1.5 rounded-full transition-all duration-300"
                        style={{ width: `${utilizationRate.doctors}%` }}
                      ></div>
                    </div>
                  </div>
                )}
              </div>
            </div>
          </div>
          
          {/* Patients Card */}
          <div className={`relative overflow-hidden rounded-xl p-4 transition-all duration-300 ${
            isSelected
              ? 'bg-gradient-to-br from-emerald-100 via-green-50 to-teal-100 border-2 border-emerald-200/50 shadow-lg'
              : 'bg-gradient-to-br from-gray-50 via-white to-green-50/30 border border-gray-200 hover:border-green-300 hover:shadow-md group-hover:bg-gradient-to-br group-hover:from-emerald-50 group-hover:via-white group-hover:to-green-50'
          }`}>
            <div className="absolute inset-0 bg-gradient-to-br from-white/60 via-transparent to-transparent"></div>
            <div className="relative flex items-center">
              <div className={`p-2 rounded-lg ${
                isSelected ? 'bg-emerald-600 text-white' : 'bg-emerald-100 text-emerald-600 group-hover:bg-emerald-600 group-hover:text-white'
              } transition-all duration-300`}>
                <UserCheck className="h-5 w-5" />
              </div>
              <div className="ml-3 flex-1">
                <p className={`text-xs font-semibold uppercase tracking-wide ${
                  isSelected ? 'text-emerald-600' : 'text-gray-500 group-hover:text-emerald-600'
                }`}>
                  Pacientes
                </p>
                <div className="flex items-baseline space-x-1">
                  <p className={`text-2xl font-black ${
                    isSelected ? 'text-emerald-800' : 'text-gray-900 group-hover:text-emerald-800'
                  }`}>
                    {group.patient_count || 0}
                  </p>
                  {group.max_patients && (
                    <p className="text-sm text-gray-500">
                      /{group.max_patients}
                    </p>
                  )}
                </div>
                {utilizationRate.patients && (
                  <div className="mt-2">
                    <div className="flex items-center justify-between mb-1">
                      <span className="text-xs text-gray-500">Ocupação</span>
                      <span className="text-xs font-semibold text-emerald-600">{utilizationRate.patients}%</span>
                    </div>
                    <div className="w-full bg-gray-200 rounded-full h-1.5">
                      <div
                        className="bg-gradient-to-r from-emerald-500 to-teal-500 h-1.5 rounded-full transition-all duration-300"
                        style={{ width: `${utilizationRate.patients}%` }}
                      ></div>
                    </div>
                  </div>
                )}
              </div>
            </div>
          </div>
        </div>

        {/* Enhanced Group Info */}
        <div className={`rounded-xl p-4 border-t ${
          isSelected
            ? 'bg-gradient-to-r from-purple-50 via-pink-50 to-blue-50 border-purple-200/30'
            : 'bg-gradient-to-r from-gray-50 via-white to-gray-50 border-gray-200 group-hover:from-purple-50 group-hover:via-pink-50 group-hover:to-blue-50'
        }`}>
          <div className={`flex items-center text-sm ${
            isSelected ? 'text-purple-700' : 'text-gray-600 group-hover:text-purple-700'
          }`}>
            <Calendar className="h-4 w-4 mr-3" />
            <span className="font-medium">
              Criado em {format(new Date(group.created_at), "dd 'de' MMMM, yyyy", { locale: ptBR })}
            </span>
          </div>
          
          {/* Limits Display */}
          <div className="grid grid-cols-2 gap-4 mt-3">
            <div className={`text-center py-2 px-3 rounded-lg border ${
              isSelected
                ? 'bg-blue-100 border-blue-200 text-blue-700'
                : 'bg-gray-100 border-gray-200 text-gray-600 group-hover:bg-blue-100 group-hover:border-blue-200 group-hover:text-blue-700'
            }`}>
              <p className="text-xs font-semibold">Limite Médicos</p>
              <p className="font-bold">{group.max_members || '∞'}</p>
            </div>
            <div className={`text-center py-2 px-3 rounded-lg border ${
              isSelected
                ? 'bg-emerald-100 border-emerald-200 text-emerald-700'
                : 'bg-gray-100 border-gray-200 text-gray-600 group-hover:bg-emerald-100 group-hover:border-emerald-200 group-hover:text-emerald-700'
            }`}>
              <p className="text-xs font-semibold">Limite Pacientes</p>
              <p className="font-bold">{group.max_patients || '∞'}</p>
            </div>
          </div>
        </div>
        
        {/* Enhanced Action Button */}
        <div className="pt-4">
          <Button
            asChild
            variant={isSelected ? "default" : "outline"}
            size="lg"
            className={`w-full justify-center font-semibold transition-all duration-300 ${
              isSelected
                ? "bg-gradient-to-r from-blue-600 via-indigo-600 to-purple-600 hover:from-blue-700 hover:via-indigo-700 hover:to-purple-700 text-white border-0 shadow-lg hover:shadow-xl"
                : "border-2 border-gray-300 hover:border-transparent hover:bg-gradient-to-r hover:from-blue-600 hover:via-indigo-600 hover:to-purple-600 hover:text-white hover:shadow-lg"
            }`}
          >
            <Link href={`/dashboard-doctor/groups/${group.id}`} className="flex items-center">
              Ver Detalhes Completos
              <ChevronRight className="h-5 w-5 ml-2" />
            </Link>
          </Button>
        </div>
        
        {/* Enhanced Selection Indicator */}
        {isSelected && (
          <div className="mt-4 p-4 bg-gradient-to-r from-blue-100 via-indigo-100 to-purple-100 rounded-xl border-2 border-blue-200/50">
            <div className="flex items-center justify-center">
              <div className="w-6 h-6 bg-gradient-to-r from-blue-500 to-purple-500 rounded-full flex items-center justify-center mr-3">
                <div className="w-3 h-3 bg-white rounded-full animate-pulse"></div>
              </div>
              <p className="text-sm font-bold text-transparent bg-clip-text bg-gradient-to-r from-blue-700 to-purple-700">
                ✓ Grupo Ativo - Pronto para gerenciamento
              </p>
            </div>
          </div>
        )}
      </CardContent>
    </Card>
  );
};