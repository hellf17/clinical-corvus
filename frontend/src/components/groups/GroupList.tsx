'use client';

import React, { useState, useEffect, useCallback } from 'react';
import { GroupWithCounts } from '@/types/group';
import { listGroups } from '@/services/groupService';
import { GroupCard } from './GroupCard';
import { Button } from '@/components/ui/Button';
import { Input } from '@/components/ui/Input';
import { Search, Plus, Shield, Users } from 'lucide-react';
import { useRouter } from 'next/navigation';
import { Spinner } from '@/components/ui/Spinner';
import { Alert, AlertDescription } from '@/components/ui/Alert';
import { useToast } from '@/hooks/use-toast';

interface GroupListProps {
  onCreateGroup?: () => void;
  selectedGroupId?: string;
  onGroupSelect?: (groupId: string) => void;
}

export const GroupList: React.FC<GroupListProps> = ({ onCreateGroup, selectedGroupId, onGroupSelect }) => {
  const [groups, setGroups] = useState<GroupWithCounts[]>([]);
  const [loading, setLoading] = useState(true);
  const [error, setError] = useState<string | null>(null);
  const [searchTerm, setSearchTerm] = useState('');
  const router = useRouter();
  const { toast } = useToast();

  const fetchGroups = useCallback(async () => {
    try {
      setLoading(true);
      setError(null);
      const response = await listGroups(searchTerm || undefined);
      setGroups(response.items);
    } catch (err: any) {
      let errorMessage = err.response?.data?.error || 'Failed to load groups';
      
      // Handle specific authentication errors
      if (err.response?.status === 401) {
        errorMessage = 'Sua sessão expirou. Por favor, faça login novamente.';
      } else if (err.response?.status === 403) {
        errorMessage = 'Você não tem permissão para acessar os grupos.';
      }
      
      setError(errorMessage);
      console.error('Error fetching groups:', err);
      
      // Show toast notification for error
      toast({
        title: "Erro ao carregar grupos",
        description: errorMessage,
        variant: "destructive",
      });
      
      // If it's an authentication error, redirect to login
      if (err.response?.status === 401) {
        setTimeout(() => {
          router.push('/sign-in');
        }, 3000);
      }
    } finally {
      setLoading(false);
    }
  }, [searchTerm, toast, router]);

  useEffect(() => {
    // Debounce search to avoid excessive API calls
    const timer = setTimeout(() => {
      fetchGroups();
    }, searchTerm ? 500 : 0); // 500ms delay for searches, immediate for initial load

    return () => clearTimeout(timer);
  }, [fetchGroups, searchTerm]);

  const handleCreateGroup = () => {
    try {
      if (onCreateGroup) {
        onCreateGroup();
      }
    } catch (err: any) {
      const errorMessage = err.response?.data?.error || 'Failed to create group';
      console.error('Error creating group:', err);
      
      // Show toast notification for error
      toast({
        title: "Erro ao criar grupo",
        description: errorMessage,
        variant: "destructive",
      });
    }
  };

  if (loading) {
    return (
      <div className="flex justify-center items-center h-64" role="status" aria-label="Carregando grupos">
        <Spinner className="h-8 w-8" aria-hidden="true" />
      </div>
    );
  }

  return (
    <div className="space-y-6">
      <div className="flex flex-col md:flex-row justify-between items-start md:items-center gap-4">
        <h2 className="text-xl font-semibold text-gray-900 dark:text-white">Seus Grupos</h2>
        <Button
          onClick={handleCreateGroup}
          className="flex items-center gap-2 w-full md:w-auto bg-gradient-to-r from-blue-600 to-purple-600 hover:from-blue-700 hover:to-purple-700"
          aria-label="Criar novo grupo"
        >
          <Plus className="h-4 w-4" aria-hidden="true" />
          Criar Grupo
        </Button>
      </div>

      <div className="relative">
        <Search className="absolute left-3 top-1/2 transform -translate-y-1/2 text-gray-400 h-4 w-4" />
        <Input
          type="text"
          placeholder="Buscar grupos por nome ou descrição..."
          value={searchTerm}
          onChange={(e) => setSearchTerm(e.target.value)}
          className="pl-10 w-full md:w-96 shadow-sm border-gray-300 focus:ring-2 focus:ring-blue-500 focus:border-blue-500"
          aria-label="Buscar grupos"
        />
      </div>

      {error && (
        <Alert
          variant="destructive"
          className="flex items-center"
          role="alert"
          aria-live="assertive"
        >
          <Shield className="h-4 w-4 mr-2" aria-hidden="true" />
          <AlertDescription>{error}</AlertDescription>
        </Alert>
      )}

      {groups.length === 0 ? (
        <div
          className="text-center py-12"
          role="region"
          aria-label={searchTerm ? "Nenhum grupo encontrado" : "Você ainda não tem grupos"}
        >
          <div className="mx-auto w-24 h-24 rounded-full bg-blue-50 dark:bg-blue-900/20 flex items-center justify-center mb-6">
            <Users className="h-12 w-12 text-blue-500 dark:text-blue-400" aria-hidden="true" />
          </div>
          <h3 className="text-lg font-medium text-gray-900 dark:text-white mb-2">
            {searchTerm ? 'Nenhum grupo encontrado' : 'Você ainda não tem grupos'}
          </h3>
          <p className="text-gray-500 dark:text-gray-400 mb-6">
            {searchTerm
              ? 'Tente ajustar sua busca ou navegar por todos os grupos.'
              : 'Crie um novo grupo para começar a colaborar com outros profissionais de saúde.'}
          </p>
          {!searchTerm && (
            <Button onClick={handleCreateGroup} className="mt-4">
              <Plus className="mr-2 h-4 w-4" aria-hidden="true" />
              Criar seu primeiro grupo
            </Button>
          )}
        </div>
      ) : (
        <div
          className="grid grid-cols-1 md:grid-cols-2 xl:grid-cols-3 gap-6"
          role="region"
          aria-label="Lista de grupos"
        >
          {groups.map((group) => (
            <GroupCard 
              key={group.id} 
              group={group}
              isSelected={selectedGroupId === group.id.toString()}
              onSelect={onGroupSelect}
            />
          ))}
        </div>
      )}
    </div>
  );
};