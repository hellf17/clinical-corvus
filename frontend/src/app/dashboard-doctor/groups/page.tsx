'use client';

import React, { useState } from 'react';
import { GroupList } from '@/components/groups/GroupList';
import { Users, UserPlus, FileText, Settings, Plus, Lightbulb, Sparkles } from 'lucide-react';
import { Card, CardContent, CardHeader, CardTitle } from '@/components/ui/Card';
import { Button } from '@/components/ui/Button';
import Link from 'next/link';
import {
  Dialog,
  DialogContent,
  DialogHeader,
  DialogTitle,
} from "@/components/ui/Dialog";
import { GroupForm } from '@/components/groups/GroupForm';

export default function GroupsPage() {
  const [selectedGroup, setSelectedGroup] = useState<string | null>(null);
  const [isCreateModalOpen, setIsCreateModalOpen] = useState(false);

  return (
    <div className="w-full max-w-7xl mx-auto space-y-8">
      {/* Page Header */}
      <div className="flex flex-col sm:flex-row sm:items-center sm:justify-between">
        <div>
          <h1 className="text-3xl font-bold text-gray-900 mb-2">Gestão de Grupos</h1>
          <p className="text-gray-600">Organize e colabore com equipes multidisciplinares</p>
        </div>
        <div className="mt-4 sm:mt-0">
          <Button
            onClick={() => setIsCreateModalOpen(true)}
            className="bg-gradient-to-r from-blue-600 to-purple-600 hover:from-blue-700 hover:to-purple-700 text-white shadow-lg"
          >
            <Plus className="h-4 w-4 mr-2" />
            Novo Grupo
          </Button>
        </div>
      </div>

      {/* Quick Actions Row */}
      <div className="grid grid-cols-1 sm:grid-cols-2 lg:grid-cols-3 gap-4 mb-8">
        {/* Manage Members Action */}
        <Button
          className={`h-auto p-4 flex flex-col items-center justify-center space-y-2 transition-all duration-200 ${
            selectedGroup
              ? 'border-white text-white hover:bg-white/10 shadow-lg'
              : 'bg-gray-100 text-gray-400 cursor-not-allowed shadow-none'
          }`}
          onClick={() => {
            if (selectedGroup) {
              window.location.href = `/dashboard-doctor/groups/${selectedGroup}/members`;
            } else {
              alert('Selecione um grupo primeiro para gerenciar membros');
            }
          }}
          disabled={!selectedGroup}
        >
          <Users className="h-5 w-5" />
          <div className="text-center">
            <div className="font-semibold text-sm">Gerenciar Membros</div>
            <div className="text-xs opacity-80">
              {selectedGroup ? 'Adicionar membros' : 'Selecione um grupo'}
            </div>
          </div>
        </Button>
        
        {/* Assign Patients Action */}
        <Button
          className={`h-auto p-4 flex flex-col items-center justify-center space-y-2 transition-all duration-200 ${
            selectedGroup
              ? 'border-white text-white hover:bg-white/10 shadow-lg'
              : 'bg-gray-100 text-gray-400 cursor-not-allowed shadow-none'
          }`}
          onClick={() => {
            if (selectedGroup) {
              window.location.href = `/dashboard-doctor/groups/${selectedGroup}/patients`;
            } else {
              alert('Selecione um grupo primeiro para atribuir pacientes');
            }
          }}
          disabled={!selectedGroup}
        >
          <FileText className="h-5 w-5" />
          <div className="text-center">
            <div className="font-semibold text-sm">Atribuir Pacientes</div>
            <div className="text-xs opacity-80">
              {selectedGroup ? 'Associar pacientes' : 'Selecione um grupo'}
            </div>
          </div>
        </Button>

        {/* Settings Action (placeholder for future use) */}
        <Button
          className={`h-auto p-4 flex flex-col items-center justify-center space-y-2 transition-all duration-200 ${
            selectedGroup
              ? 'border-white text-white hover:bg-white/10 shadow-lg'
              : 'bg-gray-100 text-gray-400 cursor-not-allowed shadow-none'
          }`}
          onClick={() => {
            if (selectedGroup) {
              window.location.href = `/dashboard-doctor/groups/${selectedGroup}/settings`;
            } else {
              alert('Selecione um grupo primeiro para acessar configurações');
            }
          }}
          disabled={!selectedGroup}
        >
          <Settings className="h-5 w-5" />
          <div className="text-center">
            <div className="font-semibold text-sm">Configurações</div>
            <div className="text-xs opacity-80">
              {selectedGroup ? 'Gerenciar grupo' : 'Selecione um grupo'}
            </div>
          </div>
        </Button>
      </div>

      {/* Main Content - Full Width Group List */}
      <div className="grid grid-cols-1 lg:grid-cols-4 gap-8">
        {/* Group List - Full Width */}
        <div className="lg:col-span-3">
          <Card className="shadow-lg border-0 bg-white/95 backdrop-blur-sm">
            <CardHeader className="border-b border-gray-100">
              <div className="flex items-center justify-between">
                <CardTitle className="text-xl flex items-center">
                  <Users className="h-6 w-6 mr-3 text-blue-600" />
                  Seus Grupos
                </CardTitle>
                <div className="text-sm text-gray-500">
                  {selectedGroup ? 'Grupo selecionado' : 'Selecione um grupo para ativar as ações'}
                </div>
              </div>
            </CardHeader>
            <CardContent className="p-6">
              <GroupList
                selectedGroupId={selectedGroup || undefined}
                onGroupSelect={setSelectedGroup}
                onCreateGroup={() => setIsCreateModalOpen(true)}
              />
            </CardContent>
          </Card>
        </div>

        {/* Tips Card - Sidebar */}
        <div className="lg:col-span-1">
          <Card className="shadow-lg border-0 bg-white/95 backdrop-blur-sm">
            <CardHeader className="pb-4">
              <CardTitle className="text-lg flex items-center">
                <Lightbulb className="h-5 w-5 mr-2 text-amber-600" />
                Dicas Úteis
              </CardTitle>
            </CardHeader>
            <CardContent>
              <div className="space-y-4">
                <div className="flex items-start">
                  <div className="flex-shrink-0 w-2 h-2 bg-blue-500 rounded-full mt-2 mr-3"></div>
                  <span className="text-sm text-gray-700">Organize grupos por <strong>especialidade</strong> ou departamento</span>
                </div>
                <div className="flex items-start">
                  <div className="flex-shrink-0 w-2 h-2 bg-green-500 rounded-full mt-2 mr-3"></div>
                  <span className="text-sm text-gray-700">Convide colegas para <strong>colaborar</strong> no cuidado</span>
                </div>
                <div className="flex items-start">
                  <div className="flex-shrink-0 w-2 h-2 bg-purple-500 rounded-full mt-2 mr-3"></div>
                  <span className="text-sm text-gray-700">Atribua pacientes para <strong>acompanhamento</strong> em equipe</span>
                </div>
                <div className="p-3 bg-gradient-to-r from-blue-50 to-purple-50 rounded-lg border border-blue-200">
                  <div className="flex items-start">
                    <div className="flex-shrink-0 w-2 h-2 bg-blue-600 rounded-full mt-2 mr-3"></div>
                    <span className="text-sm font-medium text-blue-800">
                      Clique em um grupo na lista para ativar as ações
                    </span>
                  </div>
                </div>
              </div>
            </CardContent>
          </Card>
        </div>
      </div>
      
      {/* Create Group Modal */}
      <Dialog open={isCreateModalOpen} onOpenChange={setIsCreateModalOpen}>
        <DialogContent className="max-w-md">
          <DialogHeader>
            <DialogTitle>Criar Novo Grupo</DialogTitle>
          </DialogHeader>
          <div className="py-4">
            <GroupForm
              onSuccess={() => {
                setIsCreateModalOpen(false);
                // Refresh the group list
                window.location.reload();
              }}
              onCancel={() => setIsCreateModalOpen(false)}
            />
          </div>
        </DialogContent>
      </Dialog>
    </div>
  );
}