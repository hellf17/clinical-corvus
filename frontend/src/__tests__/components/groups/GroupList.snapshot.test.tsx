import React from 'react';
import { render } from '@testing-library/react';
import { GroupList } from '@/components/groups/GroupList';
import { Group } from '@/types/group';

// Mock the group service
jest.mock('@/services/groupService', () => ({
  listGroups: jest.fn(),
}));

// Mock Next.js router
jest.mock('next/navigation', () => ({
  useRouter: () => ({
    push: jest.fn(),
  }),
}));

// Mock the GroupCard component
jest.mock('@/components/groups/GroupCard', () => ({
  GroupCard: ({ group }: { group: Group }) => (
    <div data-testid="group-card-mock">Mock Group Card: {group.name}</div>
  ),
}));

describe('GroupList Snapshot Tests', () => {
  const mockGroups: Group[] = [
    {
      id: 1,
      name: 'Test Group 1',
      description: 'First test group',
      created_at: '2023-01-01T00:00:00Z',
      updated_at: '2023-01-01T00:00:00Z',
    },
    {
      id: 2,
      name: 'Test Group 2',
      description: 'Second test group',
      created_at: '2023-01-02T00:00:00Z',
      updated_at: '2023-01-02T00:00:00Z',
    },
  ];

  beforeEach(() => {
    jest.clearAllMocks();
  });

  it('matches snapshot with groups', async () => {
    const { listGroups } = require('@/services/groupService');
    listGroups.mockResolvedValue({ items: mockGroups });
    
    const { asFragment } = render(<GroupList />);
    
    // Wait for the component to load data
    await new Promise(resolve => setTimeout(resolve, 0));
    
    expect(asFragment()).toMatchSnapshot();
  });

  it('matches snapshot with empty groups', async () => {
    const { listGroups } = require('@/services/groupService');
    listGroups.mockResolvedValue({ items: [] });
    
    const { asFragment } = render(<GroupList />);
    
    // Wait for the component to load data
    await new Promise(resolve => setTimeout(resolve, 0));
    
    expect(asFragment()).toMatchSnapshot();
  });

  it('matches snapshot while loading', () => {
    const { listGroups } = require('@/services/groupService');
    listGroups.mockResolvedValue({ items: [] });
    
    const { asFragment } = render(<GroupList />);
    
    expect(asFragment()).toMatchSnapshot();
  });

  it('matches snapshot with error', async () => {
    const { listGroups } = require('@/services/groupService');
    listGroups.mockRejectedValue(new Error('Failed to load groups'));
    
    const { asFragment } = render(<GroupList />);
    
    // Wait for the component to handle the error
    await new Promise(resolve => setTimeout(resolve, 0));
    
    expect(asFragment()).toMatchSnapshot();
  });
});