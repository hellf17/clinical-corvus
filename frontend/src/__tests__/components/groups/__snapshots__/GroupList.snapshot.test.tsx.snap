// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`GroupList Snapshot Tests matches snapshot while loading 1`] = `
<DocumentFragment>
  <div
    aria-label="Carregando grupos"
    class="flex justify-center items-center h-64"
    role="status"
  >
    <div
      aria-label="loading"
      class="inline-block animate-spin rounded-full border-2 border-current border-t-transparent h-8 w-8"
      role="status"
    >
      <span
        class="sr-only"
      >
        Carregando...
      </span>
    </div>
  </div>
</DocumentFragment>
`;

exports[`GroupList Snapshot Tests matches snapshot with empty groups 1`] = `
<DocumentFragment>
  <div
    aria-label="Carregando grupos"
    class="flex justify-center items-center h-64"
    role="status"
  >
    <div
      aria-label="loading"
      class="inline-block animate-spin rounded-full border-2 border-current border-t-transparent h-8 w-8"
      role="status"
    >
      <span
        class="sr-only"
      >
        Carregando...
      </span>
    </div>
  </div>
</DocumentFragment>
`;

exports[`GroupList Snapshot Tests matches snapshot with error 1`] = `
<DocumentFragment>
  <div
    aria-label="Carregando grupos"
    class="flex justify-center items-center h-64"
    role="status"
  >
    <div
      aria-label="loading"
      class="inline-block animate-spin rounded-full border-2 border-current border-t-transparent h-8 w-8"
      role="status"
    >
      <span
        class="sr-only"
      >
        Carregando...
      </span>
    </div>
  </div>
</DocumentFragment>
`;

exports[`GroupList Snapshot Tests matches snapshot with groups 1`] = `
<DocumentFragment>
  <div
    aria-label="Carregando grupos"
    class="flex justify-center items-center h-64"
    role="status"
  >
    <div
      aria-label="loading"
      class="inline-block animate-spin rounded-full border-2 border-current border-t-transparent h-8 w-8"
      role="status"
    >
      <span
        class="sr-only"
      >
        Carregando...
      </span>
    </div>
  </div>
</DocumentFragment>
`;
